docs/messaging/gcp-pubsub.md

```diff
@@ -51,10 +58,70 @@ flowchart LR;
 | SpanKind | `PRODUCER` | `PRODUCER` | `CLIENT` |
 | Status | `Ok` | `Ok` | `Ok` |
 | `messaging.batch.message_count` |  |  | 2 |
-| `messaging.destination.name` | `"T"` | `"T"` | `"T"` |
+| `messaging.destination.template` | `"projects/P/topics/T"` | `"projects/P/topics/T"` | `"projects/P/topics/T"` |
 | `messaging.operation` | `"create"` | `"create"` | `"publish"` |
 | `messaging.message.id` | `"a1"` | `"a2"` | |
 | `messaging.message.envelope.size` | `1` | `1` | |
 | `messaging.system` | `"gcp_pubsub"` | `"gcp_pubsub"` | `"gcp_pubsub"` |
 
+### Unary Pull Example
+
+```mermaid
+flowchart TD;
+  subgraph CONSUMER
+  direction LR
+  R1[Receive m1]
+  SM1[Settle m1]
+  EM1[Extend m1]
+  end
+  subgraph PRODUCER
+  direction LR
+  CM1[Create m1]
+  PM1[Publish]
+  end
+  %% Link 0
+  CM1-. link .-PM1;
+  %% Link 1
+  CM1-. link .-R1;
+  %% Link 2
+  R1-. link .-SM1;
+  %% Link 3
+  R1-. link .-EM1;
+
+  %% Style the node and corresponding link
+  %% Producer links and nodes
+  classDef producer fill:green
+  class PM1,CM1 producer
+  linkStyle 0 color:green,stroke:green
+
+  %% Consumer links and nodes
+  classDef consumer fill:#032a61
+  class R1 consumer
+  linkStyle 1 color:#032a61,stroke:#032a61
+
+  classDef ack fill:#577eb5
+  class SM1 ack
+  linkStyle 2 color:#577eb5,stroke:#577eb5
+
+  classDef extend fill:#0560f2
+  class EM1 extend
+  linkStyle 3 color:#0560f2,stroke:#0560f2
+```
+
+| Field or Attribute | Span Create A | Span Publish A | Span Receive A | Span Extend A | Span Settle A |
+|-|-|-|-|-|-|
+| Span name | `T create` | `publish` |  `S receive` | `S extend` |`S settle` |
+| Parent |  |  |  | |  |
+| Links |  | Span Create A | Span Create A | Span Receive A | Span Receive A |
```

Comment:
Note: these point to the receive spans, NOT the create spans. This is because there could be multiple subscriptions attached to a topic and therefore multiple receive spans. This correlates to the correct span. You also can get back to the create span through the create/receive link

Reply:
Do you also attach the "Receive" context to the message? Or are there other ways you can ensure that the "Receive" context is available during the "Settle" operation?

Reply:
No, we do not inject the "Receive" context into the message attributes. Before the receive ends, we create an AckHandler, so we get the current active span ("Receive") and store it on the AckHandler (which controls the settle operation).